mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -821,6 +827,9 @@ private[spark] class FileSystemOverwrite extends Logging {
 
 private[spark] object ReadWriteUtils {
 
+  // This member is set by Spark Connect MLHandler.
+  @volatile var safeMLClassLoader: String => Class[_] = null
```

Comment:
The design pattern does not look right to me. It should be `val` or `lazy val` that is initialized once and depending on the config delegates to `Utils.classForName`.

Reply:
unfortunately `val` or `lazy val` does not work here.
because `mllib` project can't add code that has dependency to `connect` project.

but, the `safeMLClassLoader` logic is defined in the `connect` project, so that we have to generate the `safeMLClassLoader` function in `connect` side first, then mllib project code can use it.